### PR TITLE
Bump to event-listener v3.0.0

### PR DESCRIPTION
**Pull request by notgull**
Testing to make sure that `event-listener v3.0.0` actually works before we release smol-rs/event-listener#58

Needs smol-rs/event-listener#60

### CODE REVIEW

**Comment by notgull** [comment]
I've modified the futures to be `!Unpin` to save a heap allocation, so this is now a breaking change.

File: src/lib.rs [hunk 1]
```diff
@@ -7,7 +7,7 @@
 //! * [`RwLock`] - a reader-writer lock, allowing any number of readers or a single writer.
 //! * [`Semaphore`] - limits the number of concurrent operations.
 
-#![warn(missing_docs, missing_debug_implementations, rust_2018_idioms)]
+#![warn(missing_debug_implementations, rust_2018_idioms)]
```
Comment (taiki-e) [inline src/lib.rs:10]:
```suggestion
#![warn(missing_docs, missing_debug_implementations, rust_2018_idioms)]
```

**Comment by taiki-e** [comment]
(If we make this a breaking change, I think we need to create a new 2.x release that includes #42 and #50 before merging this.)

**Comment by notgull** [comment]
`event-listener` v3 and `event-listener-strategy` v0.1 are now released, so we are fine to review this PR.

File: src/semaphore.rs [hunk 2]
```diff
@@ -225,7 +222,9 @@ pub struct AcquireArc {
     semaphore: Arc<Semaphore>,
 
     /// The listener waiting on the semaphore.
-    listener: Option<EventListener>,
+    ///
+    /// TODO: At the next breaking release, remove the `Pin<Box<>>` and make this type `!Unpin`.
+    listener: Option<Pin<Box<EventListener>>>,
```
Comment (taiki-e) [inline src/semaphore.rs:227]: Was this missed when resolving other TODOs?

File: src/mutex.rs [hunk 3]
```diff
@@ -259,158 +260,192 @@ impl<T: Default + ?Sized> Default for Mutex<T> {
...
+    pub(crate) wait();
+}
 
-    /// We are acquiring the mutex through the slow path.
-    AcquireSlow(AcquireSlow<Arc<Mutex<T>>, T>),
+pin_project_lite::pin_project! {
+    #[project = LockArcInnardsProj]
+    enum LockArcInnards<T: ?Sized> {
+        /// We have not tried to poll the fast path yet.
+        Unpolled { mutex: Option<Arc<Mutex<T>>> },
```
Comment (taiki-e) [inline src/mutex.rs:332]: FYI, I think by using project_replace, we can eliminate the need for Option here and use the same way as before.

**Review by taiki-e — COMMENTED** [review]
LGTM. Could you update the CI config to test cases where the default features are disabled?

**Review by taiki-e — APPROVED** [review]

**Pull request merged**